### PR TITLE
Weight watchers

### PR DESCRIPTION
**Pull request by AGuyNamedTate**
Fixed all conflicts and moved a few methods around. There were a few faulty calls and duplicate lines in the previous merge so those have also been fixed. Fixed all requests and reformatted the document so the linter is happy. I also changed most variable names to follow python conventions

### CODE REVIEW

File: app/parseBoard.py [hunk 1]
```diff
@@ -106,14 +107,52 @@ def bullyPathing(converted_data, pathBoard):
...
+
+    Args:
+        converted_data (json): python readable json
+        path (list): path from a*
+
+    Return:
+        updated board (list) with potentially new values around LARGER snakes heads
+    """
+    me = converted_data["you"]["id"]
+    board = pathBoard
```
Comment (kylebisley) [inline app/parseBoard.py]: Suggest you just change the parameter to reflect this name preference

**Review by kylebisley — DISMISSED** [review]
Sorry Tate, to sleepy to finish. Will look again tomorrow.

File: app/parseBoard.py [hunk 2]
```diff
@@ -106,14 +107,52 @@ def bullyPathing(converted_data, pathBoard):
...
+
+
+def cowardPathing(converted_data, pathBoard):
+    """
+    Finds Snakes that are LARGER than us and assigns the area around their head with the LARGER_Snake_Future_Head
+
+    May need logical update for snake bodies but it should be fine
+
+    Args:
+        converted_data (json): python readable json
```
Comment (DimaMzk) [inline app/parseBoard.py]: (list) not (json)

**Review by DimaMzk — CHANGES_REQUESTED** [review]
I like the change from magic numbers to Constants 👍

File: app/parseBoard.py (hunk 1 above)
Comment (DimaMzk) [inline app/parseBoard.py]: Why are we dealing with board and pathBoard as two seperate reference variables to the same thing? If you meant to copy it. `Import copy`, `copy.deepcopy(list)`

File: app/parseBoard.py (hunk 2 above)
Comment (kylebisley) [reply app/parseBoard.py]: Wait really? We've been calling this Json in some other documentation.... I thought it was a dictionary not a list.

Comment (DimaMzk) [reply app/parseBoard.py]: *(dict) not (json) my bad 😂

Comment (DimaMzk) [reply app/parseBoard.py]: Json isn't a valid python type. Python Documentation requires a valid python types

File: app/main.py [hunk 3]
```diff
@@ -89,7 +85,6 @@ def end():
     return end_response()
 
 
-
 def boardToArray(dataDump):
```
Comment (kylebisley) [inline app/main.py]: So I'm not sure how you've done it but this function has already been moved to parsableBoard and now is back in main.py 

Easiest solution I see is to copy paste this one overtop of the the parsableBoard representation of it (if you've made changes) or just simply remove the code from main.py

File: app/parseBoard.py (hunk 2 above)
Comment (kylebisley) [reply app/parseBoard.py]: Ok. Well that's a change that need to be reflected other places as well.

File: app/parseBoard.py (hunk 1 above)
Comment (kylebisley) [reply app/parseBoard.py]: Do you have to use deepcopy to protect the original? Like is python actually doing a yeold pointer copy in Tates code? I thought that python was high enough level that it had abstracted away from the pointer copy unless you explicitly wanted to...

File: app/parseBoard.py (hunk 2 above)
Comment (AGuyNamedTate) [reply app/parseBoard.py]: Ill jump on it in a bit

File: app/parseBoard.py (hunk 1 above)
Comment (DimaMzk) [reply app/parseBoard.py]: Python just gives the reference. Any changes to one happens to the other

Comment (AGuyNamedTate) [reply app/parseBoard.py]: So in any place where we pass a parameter and assign it locally, we should instead only use the parameter that was passed? This makes sense if python is pass by reference...

Comment (kylebisley) [reply app/parseBoard.py]: Also means we can skip returning a bunch of stuff if this is true.

File: app/main.py [hunk 4]
```diff
@@ -135,140 +130,68 @@ def boardToArray(dataDump):
...
-                board[y][x] = OPEN_SPACE
-    return board
-
-
-def getMinPathToFood(converted_data, pathBoard):
+def getMinPathToFood(converted_data, path_board):
     """
     Checks for lightest path to food.
     Args:
         converted_data (json): converted python representation of current game
```
Comment (DimaMzk) [inline app/main.py]:
```suggestion
        converted_data (dict): converted python representation of current game
```

File: app/main.py [hunk 5]
```diff
@@ -135,140 +130,68 @@ def boardToArray(dataDump):
...
 
                                 snapshot
-        pathBoard (int array): integer representation of board
+        path_board (int array): integer representation of board
```
Comment (DimaMzk) [inline app/main.py]:
```suggestion
        path_board (list): integer representation of board
```

File: app/main.py [hunk 6]
```diff
@@ -135,140 +130,68 @@ def boardToArray(dataDump):
...
     Returns:
-        shortestPath (path): shortest path to food
+        shortest_path (path): shortest path to food
         OR
-        shortestPath (string): "Unassigned" when it can't path to food
+        shortest_path (string): "Unassigned" when it can't path to food
```
Comment (DimaMzk) [inline app/main.py]:
```suggestion
        shortest_path (str): "Unassigned" when it can't path to food
```

File: app/main.py [hunk 7]
```diff
@@ -135,140 +130,68 @@ def boardToArray(dataDump):
...
     Checks for lightest path to food.
     Args:
         converted_data (json): converted python representation of current game
 
                                 snapshot
-        pathBoard (int array): integer representation of board
+        path_board (int array): integer representation of board
     Returns:
-        shortestPath (path): shortest path to food
+        shortest_path (path): shortest path to food
```
Comment (DimaMzk) [inline app/main.py]:
```suggestion
        shortest_path (list): shortest path to food
```

**Review by DimaMzk — CHANGES_REQUESTED** [review]

**Review by DimaMzk — APPROVED** [review]

**Pull request merged**